native/fluent_native/src/lib.rs

```diff
@@ -79,73 +83,82 @@ fn format_pattern<'a>(
+                },
+                TermType::Number => match value.decode::<f64>() {
                     Ok(float) => FluentValue::from(float),
-                    Err(_e) => {
-                        match value.decode::<i64>() {
-                            Ok(integer) => FluentValue::from(integer),
-                            Err(_e) => panic!("Mismatched types")
-                        }
-                    }
-                }
-            }
-            _ => panic!("Mismatched types") // TODO: Add error handling
-        };
-        (format!("{:?}", key), fluent_value)
-    }).collect();
+                    Err(_e) => match value.decode::<i64>() {
+                        Ok(integer) => FluentValue::from(integer),
+                        Err(_e) => panic!("Mismatched types"),
+                    },
+                },
+                _ => panic!("Mismatched types"), // TODO: Add error handling
+            };
+            (format!("{:?}", key), fluent_value)
+        })
+        .collect();
 
     // Locking bundle to write
     let bundle = container.data.read().unwrap();
 
     // Getting message
     let msg = match bundle.get_message(&msg_id) {
         Some(msg) => msg,
-        None => return Ok((atoms::error(), atoms::bad_msg()).encode(env))
+        None => return Ok((atoms::error(), atoms::bad_msg()).encode(env)),
     };
 
     // Getting args
     let mut args = FluentArgs::new();
     for (key, value) in &arg_ids {
-        args.insert(key, value.clone());
+        args.set(key, value.clone());
     }
 
     // Getting errors
     let mut errors = vec![];
 
     // Formatting pattern
-    let pattern = match msg.value {
+    let pattern = match msg.value() {
         Some(value) => value,
-        None => return Ok((atoms::error(), atoms::no_value()).encode(env))
+        None => return Ok((atoms::error(), atoms::no_value()).encode(env)),
     };
 
-    let value = bundle.format_pattern(&pattern, Some(&args), &mut errors);
+    let value = bundle.format_pattern(pattern, Some(&args), &mut errors);
 
     Ok((atoms::ok(), value.into_owned()).encode(env))
 }
 
 #[rustler::nif]
-fn assert_locale<'a>(env: Env<'a>, lang: String) -> Result<Term<'a>, Error> {
+fn assert_locale(env: Env<'_>, lang: String) -> Result<Term<'_>, Error> {
     // Getting language
     match lang.parse::<LanguageIdentifier>() {
         Ok(lang_id) => Ok((atoms::ok(), &lang_id.to_string()).encode(env)),
-        Err(_e) => Ok((atoms::error(), (atoms::bad_locale(), lang)).encode(env))
+        Err(_e) => Ok((atoms::error(), (atoms::bad_locale(), lang)).encode(env)),
     }
 }
 
 rustler::init!(
     "Elixir.Fluent.Native",
     [init, with_resource, format_pattern, assert_locale],
     load = on_init
-);
\ No newline at end of file
+);
+
+#[cfg(test)]
+mod test {
+    use crate::*;
+
+    fn assert_send_and_sync<T: Send + Sync + 'static>() {}
+
+    #[test]
+    fn it_is_send_and_sync() {
+        assert_send_and_sync::<Container>();
```

Comment:
Why do we need to check this with test? If this traits are required somewhere - then compiler must check it in place, innit?

Reply:
True, it's just something I always add to a test suite to make it explicit that this type MUST implement it. Without this, code readers need to go check the trait bounds required by Rustler to know that Container must be Send + Sync